### PR TITLE
added code for create designation

### PR DESCRIPTION
**Pull request by anilkumarkammalapalli**

### CODE REVIEW

File: src/main/java/com/igot/cb/designation/service/impl/DesignationServiceImpl.java [hunk 1]
```diff
@@ -249,5 +348,136 @@ private Date parseDate(String value) throws Exception {
     SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
     return dateFormat.parse(value);
   }
+  public ApiResponse readDesignation(String Id) {
+    ApiResponse response = new ApiResponse();
+    try {
+      StringBuilder strUrl = new StringBuilder(cbProperties.getKnowledgeMS());
+      strUrl.append(cbProperties.getOdcsTermCrete()).append("/").append(Id).append("?framework=")
```
Comment (karthik-tarento) [inline src/main/java/com/igot/cb/designation/service/impl/DesignationServiceImpl.java]: why are we using cbProperties.getOdcsTermCreate() method here ??
it suppose to read right ?

File: src/main/java/com/igot/cb/designation/service/impl/DesignationServiceImpl.java [hunk 2]
```diff
@@ -112,6 +122,95 @@ public void loadDesignationFromExcel(MultipartFile file) {
...
+          response.getParams().setStatus(Constants.FAILED);
+        } else if (HttpStatus.NOT_FOUND.equals(readResponse.getResponseCode())) {
+          Map<String, Object> reqBody = new HashMap<>();
+          reqBody.put(Constants.CODE, ref_Id);
+          reqBody.put(Constants.REF_ID, ref_Id);
+          reqBody.put(Constants.REF_TYPE, ref_Type);
+          reqBody.put(Constants.NAME, name);
+          Map<String, Object> parentObj = new HashMap<>();
+          parentObj.put(Constants.IDENTIFIER,
+                  cbProperties.getOdcsFrameworkName() + "_" + cbProperties.getOdcsCategoryName());
```
Comment (karthik-tarento) [inline src/main/java/com/igot/cb/designation/service/impl/DesignationServiceImpl.java]: both of this property looks like common one... instead these should be named with odcsDesignationFramework and odcsDesignationCategory

File: src/main/java/com/igot/cb/designation/service/impl/DesignationServiceImpl.java [hunk 3]
```diff
@@ -112,6 +122,95 @@ public void loadDesignationFromExcel(MultipartFile file) {
...
+          response.getParams().setStatus(Constants.FAILED);
+        }
+      } else {
+        //if desg. is not active.
+        log.error("Failed to create Designation exists with name: " + ref_Id);
+        response.getParams().setErr("Failed to create Designation.");
+        response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+        response.getParams().setStatus(Constants.FAILED);
+      }
+    } else
```
Comment (karthik-tarento) [inline src/main/java/com/igot/cb/designation/service/impl/DesignationServiceImpl.java]: Code is not formatted properly

File: src/main/java/com/igot/cb/designation/service/impl/DesignationServiceImpl.java [hunk 4]
```diff
@@ -112,6 +122,95 @@ public void loadDesignationFromExcel(MultipartFile file) {
         });
     log.info("DesignationServiceImpl::loadDesignationFromExcel::created the designations");
   }
+  @Override
+  public ApiResponse createDesignation(JsonNode request){
+    ApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_DESIGNATION_CREATE);
+    payloadValidation.validatePayload(Constants.DESIGNATION_CREATE_PAYLOAD_VALIDATION,request);
```
Comment (karthik-tarento) [inline src/main/java/com/igot/cb/designation/service/impl/DesignationServiceImpl.java]: what happens when validation fails ?

File: src/main/java/com/igot/cb/pores/util/CbProperties.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+package com.igot.cb.pores.util;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+@Component
+
+public class CbProperties {
```
Comment (karthik-tarento) [inline src/main/java/com/igot/cb/pores/util/CbProperties.java]: Don't we have properties class already ??
why we need new class ?

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -93,3 +93,12 @@ sb.search.service.host=http://search-service:9000
 sb.composite.v4.search=/v4/search
 playlist_course_facets=courseCategory,primaryCategory
 elastic.required.field.designation.json.path=/EsFieldsmapping/designationEsRequiredFile.json
+
+knowledge.mv.service=http://knowledge-mw-service:5000/
+odcs.term.create=v1/framework/term/create
+odcs.framework.name=
+odcs.category.name=
```
Comment (karthik-tarento) [inline src/main/resources/application.properties]: are these used only for designation ?? is it designation master / adding designation to org ?

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

**Review by sureshece16 — APPROVED** [review]

**Pull request merged**